pkg/node/node.go

```diff
@@ -993,12 +993,29 @@ func (s *Service) nodeExpandRawBlockVolume(ctx context.Context, volumeWWN string
 	if len(deviceNames) > 0 {
 		var devName string
 		for _, deviceName := range deviceNames {
-			devicePath := sysBlock + deviceName
-			log.Infof("Rescanning unmounted (raw block) device %s to expand size", deviceName)
-			err = s.Fs.GetUtil().DeviceRescan(context.Background(), devicePath)
-			if err != nil {
-				log.Errorf("Failed to rescan device (%s) with error (%s)", devicePath, err.Error())
-				return nil, status.Error(codes.Internal, err.Error())
+			if strings.HasPrefix(deviceName, "nvme") {
+				nvmeControllerDevice, err := s.Fs.GetUtil().GetNVMeController(deviceName)
+				if err != nil {
+					log.Errorf("Failed to rescan device (%s) with error (%s)", deviceName, err.Error())
+					return nil, status.Error(codes.Internal, err.Error())
+				}
+				if nvmeControllerDevice != "" {
```

Comment:
So there seems to be a place where nvmeControllerDevice can be empty but GetNVMeController() returns no error. Is this code still then correct?

Reply:
For PowerMax, we have two devices: nvme0c0n1 and nvme0n1.
![image](https://github.com/user-attachments/assets/6e35e23d-746c-4c4a-be13-8b193e62345a)

Here, nvme0c0n1 is a controller, but nvme0n1 is not.

The GetNVMeController function processes each device as follows:
1. nvme0c0n1 - Successfully returns the controller nvme0.
2. nvme0n1 - print logger `Not a valid nvme controller device: nvme0n1` and return empty nvmeControllerDevice